app/agents/chain_of_thoughts.py

```diff
@@ -1,30 +1,45 @@
 from typing import Dict, Callable
 from andromeda_chain import AndromedaChain
 from agents.flow_based import BaseFlowAgent
-from app.flow.flow import Flow, PromptNode, ToolNode
-from prompts.guicance_cot import ChainOfThoughts
+from flow.flow import Flow, PromptNode, ToolNode, ChoiceNode
+from prompts.guicance_cot import ChainOfThoughts, PROMPT_START_STRING
 
 
 class ChainOfThoughtsAgent(BaseFlowAgent):
     def __init__(self, andromeda: AndromedaChain, tools: Dict[str, Callable[[str], str]]):
-        super().__init__(andromeda, tools, flow)
-        self.valid_tools = list(tools.keys()) + ["Reply"]
-        self.valid_answers = ["Action", "Final Answer"]
-
         def execute_tool(variables):
-            action_name = variables["action"]
-            action_input = variables["action_input"]
+            action_name = variables["tool_name"]
+            action_input = variables["actInput"]
```

Comment:
just a note on this one... I wonder why it's camel case

Reply:
It was like this before and I didn’t want to break the prompt - should fix it at some point